haproxy_agent.py

```diff
@@ -266,17 +273,26 @@ def perform_basic_auth(headers: str, username: str, password: str) -> bool:
     return False
 
 
-async def nc_get_exapp(appid: str) -> ExApp | None:
-    # TO-DO: query AppAPI endpoint for such exapp_record
-    return ExApp(
-        exapp_token="12345",
-        exapp_version="1.1.1",
-        port=23000,
-        routes=[
-            ExAppRoute(url="/http", access_level="PUBLIC"),
-            ExAppRoute(url="/ws", access_level="PUBLIC")
-        ]
-    )
+EX_APP_URL = f"{ \
+    NC_INSTANCE_URL.removesuffix('/').removesuffix('/index.php') \
+}/index.php/apps/app_api/harp/exapp-meta"
+
+class RequestFailException(Exception):
+    pass
+
+async def nc_get_exapp(app_id: str) -> ExApp | None:
+    async with aiohttp.ClientSession() as session, session.get(EX_APP_URL, headers={
+        "harp-shared-key": SHARED_KEY
+    }, params={
+        "appId": app_id
+    }) as resp:
+        if not resp.ok:
```

Comment:
let's return `None` in this function if we need to ban (i.e. AppAPI returned 404 for example - then we return `None`)

and move handling of system Exceptions (communication errors) in "async def exapps_msg"